core/src/com/treode/notify/Notification.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2014 Treode, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.treode.notify
+
+/*
+ * An object that accumulates a list of notifications (errors) and returns them
+ * to the caller in batches. This is to avoid Exception-style error reporting,
+ * where the execution flow is broken by the first mistake, and the caller must
+ * resolve errors one-by-one.
+ *
+ * Inspired by http://martinfowler.com/articles/replaceThrowWithNotification.html
+ */
+
+class Notification {
+  val list: List[Message] = List.empty;
+
+  /* Add message to the list. */
+  def add (message: Message) =
+    list :+ message
```

Comment:
This isn't actually doing anything. This line appends to an immutable list, returns a new list and then stores that result...nowhere.

    val list: List[Message] = List.empty;

    def add (message: Message) =
      list :+ message

I see this bug went unnoticed because DriveGroupSpec does not contain tests to check that DriveGroup reports errors.  So, as part of this change, you get to fill in the Spec. Lucky you. I've written up [one example test](https://github.com/Topher-the-Geek/store/commit/c4596d0f7943b6287c8955e8f9e130e8fdb77989) to help you get started.